base_phone_rate_import_bandwith/wizard/base_phone_rate_import.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2019, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import Warning
+import odoo.tools as tools
+
+from xlrd import open_workbook
+import base64
+import tempfile
+
+
+class BasePhoneRateImportBandwith(models.TransientModel):
+    _name = "base.phone.rate.import.bandwith"
+    _description = "Import Rates from Bandwidth.com"
+
+    upload_file = fields.Binary("Upload File")
+    filename = fields.Char("File name")
+
+    @api.multi
+    def import_bandwith(self):
+        res_country_obj = self.env['res.country']
+        phone_rate_obj = self.env['phone.rate']
+        for data in self:
+            data_list = []
+            filename_str = tools.ustr(data.filename)
+            if not data.upload_file:
+                raise Warning(_('Please select a file to proceed.'))
+            if not filename_str[-5:] == ".xlsx":
+                raise Warning(_('Select .xlsx file only'))
+            csv_data = base64.decodestring(data.upload_file)
+            temp_path = tempfile.gettempdir()
+            fp = open(temp_path + '/xsl_file.xlsx', 'wb+')
+            fp.write(csv_data)
+            fp.close()
+            wb = open_workbook(temp_path + '/xsl_file.xlsx')
```

Comment:
You should be able, without having to write it to a temp disk file.
Not 100% sure if xlrd supports that, though.